### PR TITLE
Add overloads for setParent(Context context).

### PR DESCRIPTION
**Pull request by carlosalberto**
This way a `Context` can be used such as:

```java
Context ctx = extractContext(...);
spanBuilder.setParent(ctx); // No need to call TracingContextUtils.getSpan()
```

### CODE REVIEW

File: api/src/main/java/io/opentelemetry/correlationcontext/CorrelationContext.java [hunk 1]
```diff
@@ -74,6 +75,28 @@
      */
     Builder setParent(CorrelationContext parent);
 
+    /**
+     * Sets the parent {@link CorrelationContext} to use from the specified {@code Context}. If no
+     * parent is provided, the value of {@link CorrelationContextManager#getCurrentContext()} at
```
Comment (jkwatson) [inline api/src/main/java/io/opentelemetry/correlationcontext/CorrelationContext.java:80]: maybe "If no parent context is available in the provided Context, the value of ...."

File: api/src/main/java/io/opentelemetry/correlationcontext/CorrelationContext.java [hunk 2]
```diff
@@ -74,6 +75,28 @@
...
+     * <p>This <b>must</b> be used to create a {@link CorrelationContext} when manual Context
+     * propagation is used.
+     *
+     * <p>If called multiple times, only the last specified value will be used.
+     *
+     * @param context the {@code Context}.
+     * @return this.
+     * @throws NullPointerException if {@code context} is {@code null}.
+     * @see #setNoParent()
+     * @since 0.4.0
```
Comment (jkwatson) [inline api/src/main/java/io/opentelemetry/correlationcontext/CorrelationContext.java:96]: 0.5.0

**Comment by jkwatson** [comment]
My main concern with this is that we're making the grpc context even more deeply embedded in the APIs. I think having the helper methods is very useful, but I'm trying to be sensitive to the desire to not expose the grpc context in our APIs.

**Comment by carlosalberto** [comment]
> My main concern with this is that we're making the grpc context even more deeply embedded in the APIs. 

Same feeling. That's why I hadn't worked on this before getting an agreement on the Context-Prop part (that and other work items ;) ). But in any case, I'd like to get things on the discussion table, for a start ;)

**Comment by jkwatson** [comment]
@carlosalberto do you want to move forward with this, even with the expectation that we'll need to revisit when we deal with grpc-context, pre GA?

**Comment by carlosalberto** [comment]
@jkwatson Yes, definitely. Had a small chat with Yuri and he mentioned he would prefer us to keep the explicit way. If/when we change `Context`, we can update this overloads.

So I will massage with code this week in hopes of having it merged ;)

**Comment by jkwatson** [comment]
@carlosalberto any update on this one?

**Comment by iNikem** [comment]
@carlosalberto friendly ping from me as well :) I am contemplating doing some experiments in instrumentation repo and I need this overload.

**Comment by carlosalberto** [comment]
Oh, my bad, totally forgot about this one. Will apply the feedback and hopefully we will be able to roll it soon ;)

**Comment by carlosalberto** [comment]
@open-telemetry/java-approvers Updated this, for your consideration ;)

File: api/src/main/java/io/opentelemetry/correlationcontext/CorrelationContext.java (hunk 1 above)
Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/correlationcontext/CorrelationContext.java]: @carlosalberto thoughts on my suggestion?

File: api/src/main/java/io/opentelemetry/correlationcontext/CorrelationContext.java (hunk 2 above)
Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/correlationcontext/CorrelationContext.java]: 0.7.0 now

File: api/src/main/java/io/opentelemetry/trace/Span.java [hunk 3]
```diff
@@ -406,6 +407,23 @@
...
+     * <p>If no {@link Span} is available in the specified {@code Context}, the resulting {@code
+     * Span} will become a root instance, as if {@link #setNoParent()} had been called.
+     *
+     * <p>If called multiple times, only the last specified value will be used. Observe that the
+     * state defined by a previous call to {@link #setNoParent()} will be discarded.
+     *
+     * @param context the {@code Context}.
+     * @return this.
+     * @throws NullPointerException if {@code context} is {@code null}.
+     * @since 0.5.0
```
Comment (jkwatson) [inline api/src/main/java/io/opentelemetry/trace/Span.java]: 0.7.0

Comment (carlosalberto) [reply api/src/main/java/io/opentelemetry/trace/Span.java]: I think there was a race condition between me pushing the changes and your review ;) In any case, all issues have been addressed.

**Pull request merged**